server/src/test/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceShardsAllocatorTests.java

```diff
@@ -231,117 +213,149 @@ public void afterPrimariesBeforeReplicas(RoutingAllocation allocation) {}
         rerouteService.assertNoPendingReroute();
     }
 
-    public void testCallListenersOnlyAfterProducingFreshInput() {
```

Comment:
This commit removes `testCallListenersOnlyAfterProducingFreshInput` and `testFailListenersOnNoLongerMasterException` (replacing them with `testConcurrency`). I think the removed tests are still valid and useful aren't they? Could we bring them back?